reVX/least_cost_xmission/offshore_layer_creator_cli.py

```diff
@@ -0,0 +1,147 @@
+
+
+@main.command
+@click.option('--vector', '-v', required=True, type=click.Path(exists=True),
+              help='Vector land mask to rasterize.')
+@click.option('--template-raster', '-t', required=True,
+              type=click.Path(exists=True),
+              help='Raster to use as a template for rasterizing the land mask')
+@click.option('--out-file', '-o', type=click.Path(),
+              help='Filename to use for rasterized land mask.')
+@click.option('--buffer', '-b', type=int,
+              help='Buffer vector features before rasterizing.')
+def rasterize_land_mask(vector: str, template_raster: str,
+                        out_file: Union[str, None],
+                        buffer: Union[int, None]):
+    """
+    Convert a vector land mask to a raster based on a template. Any features in
+    the vector layer will be assigned a 1 value in the raster, all other cells
+    will have a value of 0.
+    """
+    cr = CombineRasters(template_raster)
+    cr.create_land_mask(vector, save_tiff=True, filename=out_file,
+                        buffer_dist=buffer)
+
+
+@main.command
+@click.option('--poi-file', '-p', required=True, type=click.Path(exists=True),
+              help='File of POIs in CSV format. Each POI must have the '
+              'following fields: "POI Name", "State", "Voltage (kV)", "Lat", '
+              'and "Long". "State" may be blank. Other fields are ignored.')
+@click.option('--template-raster', '-t', required=True,
+              type=click.Path(exists=True),
+              help='Raster to extract CRS from.')
+@click.option('--out-file', '-o', type=click.Path(), required=True,
+              help='Filename to use for POI lines GeoPackage file.')
+def convert_pois(poi_file: str, template_raster: str, out_file: str):
+    """
+    Convert points of interconnection (POI) to short lines. The transmission
+    routing code requires all transmission elemnts to be lines. The POIs
+    defined in the CSV will be converted to lines and labeled as substations.
+    As all substations must be link to a transmission line, a synthetic
+    transmission line is created that is linked to the POIs.
+    """
+    convert_pois_to_lines(poi_file, template_raster, out_file)
+
+
+@main.command
+@click.option('-c', '--config', 'config_fpath', type=click.Path(exists=True),
+              required=True, help='Configuration JSON.')
+@click.option('--create-h5/--use-existing-h5', default=True, show_default=True,
+              help='Create a new offshore H5 data file or use an existing H5.')
+def from_config(config_fpath: str, create_h5: bool):
+    """
+    Create offshore barriers and frictions from a config file.
+    """
+    config = OffshoreCreatorConfig(config_fpath)
+    if create_h5 and config.ex_offshore_h5_fpath is None:
+        click.echo('ex_offshore_h5_fpath must be set unless H5 creation is '
+                   'disabled with --dont-create-h5', err=True)
+        sys.exit(1)
```

Comment:
Is this more robust than a simple return? I'm not opposed but am curious what advantages it has over a function return

Reply:
That's an error condition. `return` I believe would have an exit code 0, or success. I'm forcing an exit code of 1, or an error. Trying to follow POSIX protocol, though a user would probably never notice.